openstack/auth_env.go

```diff
@@ -22,6 +39,16 @@ func AuthOptionsFromEnv() (gophercloud.AuthOptions, error) {
 	domainID := os.Getenv("OS_DOMAIN_ID")
 	domainName := os.Getenv("OS_DOMAIN_NAME")
 
+	// If OS_PROJECT_ID is set, overwrite tenantID with the value.
+	if v := os.Getenv("OS_PROJECT_ID"); v != "" {
```

Comment:
> If Gophercloud had full support for `ProjectName` and `ProjectID` in `AuthOptions`, I could see throwing an error if both Tenant and Project were used. 

@jtopjian so gophercloud doesn't have full support for `ProjectName` and `ProjectID`. Please, would you elaborate more on what's missing?